README.md

```diff
@@ -6,10 +6,114 @@
+* description: additional information about the build.
+* id: normally number as a string.
+* timeInMillis: time since the epoch when the build was scheduled.
+* startTimeInMillis: time since the epoch when the build started running.
+* duration: duration of the build in milliseconds.
+* durationString: a human-readable representation of the build duration.
+* previousBuild: previous build of the project, or null.
+* previousBuildInProgress: previous build of the project that is currently building, or null.
+* previousBuiltBuild: previous build of the project that has been built (may be currently building), or null.
+* previousCompletedBuild: previous build of the project that has last finished building, or null.
+* previousFailedBuild: previous build of the project that has last failed to build, or null.
+* previousNotFailedBuild: previous build of the project that did not fail to build (eg. result is successful or unstable), or null.
+* previousSuccessfulBuild: previous build of the project that has successfully built, or null.
+* nextBuild: next build of the project, or null.
+* absoluteUrl: URL of build index page.
+* buildVariables: for a non-Pipeline downstream build, offers access to a map of defined build variables; for a Pipeline downstream build, any variables set globally on env at the time the build ends. Child Pipeline jobs can use this to report additional information to the parent job by setting variables in env. Note that build parameters are not shown in buildVariables.
+* changeSets: a list of changesets coming from distinct SCM checkouts; each has a kind and is a list of commits; each commit has a commitId, timestamp, msg, author, and affectedFiles each of which has an editType and path; the value will not generally be Serializable, so you may only access it inside a method marked @NonCPS.
+* upstreamBuilds: a list of upstream builds. These are the builds of the upstream projects whose artifacts feed into this build.
+* rawBuild: a hudson.model.Run with further APIs, only for trusted libraries or administrator-approved scripts outside the sandbox; the value will not be Serializable, so you may only access it inside a method marked @NonCPS.
+* keepLog: true if the log file for this build should be kept and not deleted.
+  If you do not wait, this step succeeds so long as the downstream build can be added to the queue (it will not even have been started). In that case there is currently no return value.
+
+##### Propagate error
+If enabled (default state), then the result of this step is that of the downstream build (e.g., success, unstable, failure, not built, or aborted) i.e. if the downstream build fails then the upstream build fails also. If disabled, then this step succeeds even if the downstream build is unstable, failed, etc.; use the result property of the return value as needed i.e if the downstream build fails it does not affect that of the upstream build. Example;
```

Comment:
```suggestion
If this option is enabled, this step inherits the status of the downstream build, for example:  success, unstable, failure, not built, or aborted.
If you disable this option, this step succeeds regardless of issues with the downstream build.
Use the result property of the return value as needed. Example;
```

Reply:
great stuff here as well, just wanted to clean up the text a bit and make it more focused

Reply:
Thank you, @kmartens27. All changes made